### PR TITLE
Add changelog doc going up to dask 0.6.1 (2015-07-23).

### LINKED ISSUE #1486: is there a changelog?
Is there a changelog?

If there is not there's no way to know what have changed since last update which is... disappointing!

### PR DESCRIPTION
**Pull request by dukebody**
Release notes are based on blaze-dev release emails:
https://groups.google.com/a/continuum.io/forum/#!topicsearchin/blaze-dev/release

Resolves #1486.

Issues:
 - I didn't find release notes for all releases in the mailing list. Weren't they published, or did I miss something? The gaps in the changelog look weird.
 - I was too lazy to include the links provided in the release emails to the PRs of features and all. I don't know if this is something very important but can be time consuming (and boring!). Perhaps do this only for newer releases? :)

Cheers!

### CODE REVIEW

**Comment by dukebody** [comment]
Perhaps we should add some notes to the development guidelines indicating that one has to update the changelog with every PR and the expected format?

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+bug fixes in ``dask.imperative`` and ``dask.bag``.
+
+0.7.0 / 2015-08-15
+------------------
+
+DataFrame
++++++++++
+This release includes significant bugfixes and alignment with the Pandas API.
+This has resulted both from use and from recent involvement by Pandas core
+ developers.
```
Comment (mrocklin) [inline docs/source/changelog.rst]: The extra whitespace here is causing unintentional quoting in the rendered version

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
++++++++++
+
+- Refactor ``dask.dataframe`` to include a full empty pandas dataframe as
+  metadata.  Drop the ``.columns`` attribute on Series
+- Add Series categorical accessor, series.nunique, drop the ``.columns``
+  attribute for series.
+- ``read_csv`` fixes (multi-column parse_dates, integer column names, etc. )
+- Internal changes to improve graph serialization
+
+Other
```
Comment (mrocklin) [inline docs/source/changelog.rst]: There are some sections, like this, that are inconsistently titled

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+
+Distributed
+
+- Move s3 and hdfs functionality into the dask repository
+- Adaptively oversubscribe workers for very fast tasks
+- Improve PyPy support
+- Improve work stealing for unbalanced workers
+- Scatter data efficienctly with tree-scatters
+
+General
```
Comment (mrocklin) [inline docs/source/changelog.rst]: There are some sections, like this, that are inconsistently titled.

**Comment by mrocklin** [comment]
Hooray!  This is great.  Thank you.  I agree that it would be a good idea to add a note to the development guidelines to update the changelog

**Comment by dukebody** [comment]
Updated the PR.

@mrocklin  What do you think about the release notes I didn't find in the mail archives?

**Comment by mrocklin** [comment]
If you felt very adventurous you could dive through the commit log or PRs.  We may be at the point of diminishing returns though.  I think that what you have done here achieves most of the benefit.

**Comment by dukebody** [comment]
OK!

I updated the changelog doc to include current development changes I considered appropriate. Some of the changes descriptions I wrote there are not very informative since I didn't investigate the change so thoroughly, but I assume the individual contributors can improve it in further commits.

I also included an extension stolen from https://github.com/benoitc/gunicorn/blob/4a04195975563d2f4ec28ffaa70d138beda76da4/docs/gunicorn_ext.py to point to Github issues and PRs.

**Comment by mrocklin** [comment]
Hey, that's pretty cool!

We might need to copy [the gunicorn license](https://github.com/benoitc/gunicorn/blob/4a04195975563d2f4ec28ffaa70d138beda76da4/LICENSE) into that source file

**Comment by dukebody** [comment]
PR updated!

Turned out that there is an extension bundled with sphinx, [`sphinx.ext.extlinks`](http://www.sphinx-doc.org/en/stable/ext/extlinks.html), that does exactly what we are looking for, so we don't need custom code nor extra copyright notes. :)

**Comment by mrocklin** [comment]
Hurrah!  Merging.

**Pull request merged**